### ISSUE TITLE
ExecutionFailed: key not found

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13143]
When running the following query:
```cypher
FOREACH (n0 IN [] | CREATE (x) ) MERGE ()<-[x:A]-()<-[:A]->()
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (key not found: SymbolUse(x@44(cfca569))
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.6.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
FOREACH (n0 IN [] | CREATE (x) ) MERGE ()<-[x:A]-()<-[:A]->()
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (key not found: SymbolUse(x@44(cfca569))`.

**Comment by InverseFalcon** [comment]
Definitely buggy. Even if we fix the direction of the last rel in the pattern to point only in one direction or the other, we get the same kind of error.

Simply changing the relationship type of the last rel in the pattern to a different type, like `:B`, allows it to succeed. Needs investigation and fixing.

**Comment by InverseFalcon** [comment]
Related issue:
https://github.com/neo4j/neo4j/issues/13146

**Comment by loveleif** [comment]
Thanks for the report! I've created a ticket in our internal bug tracker for further investigation ([link](https://trello.com/c/52Vy8GGo/4698-executionfailed-key-not-found)).

**Comment by Lojjs** [comment]
We have found this bug internally already so there is a fix coming up in Neo4j 5.7.0 which should be released later today or tomorrow.
